statistic: Allowing input multiple desire states

cathay4t opened the pull request:
Introducing `NetworkState::merge_desire()`, use could merge multiple
desire states and generate the statistic of the final outcome.

The `nmstatectl statistic` subcommand now support multiple desired
state files.

Manpage of `nmstatectl` updated to include this change.

Unit test case included.

Resolves: https://issues.redhat.com/browse/RHEL-21048

liangwen12year commented:
I did not see the `nmstatectl statistic` subcommand in the man page, can we add it somehow ?   Other than this, looks good to me.

cathay4t committed:
statistic: Allowing input multiple desire states

Introducing `NetworkState::merge_desire()`, use could merge multiple
desire states and generate the statistic of the final outcome.

The `nmstatectl statistic` subcommand now support multiple desired
state files.

Manpage of `nmstatectl` updated to include this change.

Unit test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Man page included.

liangwen12year commented on doc/nmstatectl.8.in:
```diff
@@ -121,6 +124,15 @@ default: \fB/etc/nmstate\fR) folder.
+.RS
+Generate statistic for specified network state files.
+Multiple desire states files are supported.
+The \fB--currrent <CURRENT_STATE_FILE>\fR argument could instust nmstate to
```
Is this a typo ? Should be "instruct" ?

ihuguet commented:
It looks like merge_desired only merges interfaces with absent state? Is that intentional?

cathay4t commented:
> It looks like merge_desired only merges interfaces with absent state? Is that intentional?

The `self.update_state()` merged other data, only interface need special treatment.
